docs/DeadmanAlert.md

```diff
@@ -0,0 +1,56 @@
+### Deadman Alert
+
+The `deadman` function is used in the context of a task to alert when a group stops reporting data.
+For example, say you have the following task definition:
+
+```
+option task = {
+    ...
+    period: 1h,
+    every:  15m,
+    ...
+}
+from(bucket: "telegraf/autogen")
+    |> range(start: -period)
+    |> filter(fn: (r) => r._measurement == "cpu")
+```
+
+In order to be alerted when the above task stops reporting data for a group, you would define a deadman alert using the `deadman` function like so:
+```
+option task = {
+    ...
+    period: 1h,
+    every:  15m,
+    ...
+}
+from(bucket: "telegraf/autogen")
+    |> range(start: -perod)
+    |> filter(fn: (r) => r._measurement == "cpu")
+    |> deadman(d: task.every)
+    |> alert(crit: (r) => true)
```

Comment:
Should `deadman` include `alert()`? If you have it include `alert()`, you don't need the hacky `(r) => true` to trigger an alert and you can also get output when the deadman returns false (meaning everything is ok) so you can get `ok` values from the alert.

Such as:

```
deadman = (d, tables=<-) => tables
    |> sort(columns: ["_time"])
    |> last()
    |> alert(crit: (r) => r._time < now() - d)
```

Reply:
Deadman should not include an alert, we do not want to couple those concepts. 

But perhaps deadman should not filter the dasta rather it should add a column `dead` to the data that indicates if it is dead or not and then the alert check can use the `dead` signal to know when a check has recovered.

```
deadman = (d, tables=<-) => 
    |> sort(columns:["_time"])
    |> last()
    |> map(fn: (r) ({r with dead: r._time < now()  -d ))
```

Then when used with a check it becomes:

```
from()
   |> range()
   |> filter()
   |> deadman()
   |> alert(crit: (r) => r.dead)
```

Reply:
To clarify, I more meant `alert()` in the way it is being used in the checks/notifications document, but that is still in revision.

So I meant more like this:

deadman = (d, tables=<-) => tables
    |> sort(columns: ["_time"])
    |> last()
    |> check(crit: (r) => r._time < now() - d)

Reply:
What I propose above is effectively the same thing as your example here except that you can still use `deadman` in other contexts besides alerting.

Reply:
Is there an example of when you do **not** want to filter out the series that are still reporting?

Reply:
Yes, in order for an alert check to recover it needs to be given the list of series that are alive as well. If we only ever send the negative signal (i.e. the series is dead), then we cannot ever determine if the series is in an OK state.